clidocstool_test.go

```diff
@@ -201,15 +203,38 @@ func TestGenAllTree(t *testing.T) {
 	require.NoError(t, err)
 	require.NoError(t, c.GenAllTree())
 
-	for _, tt := range []string{"buildx.md", "buildx_build.md", "buildx_stop.md", "docker_buildx.yaml", "docker_buildx_build.yaml", "docker_buildx_install.yaml", "docker_buildx_stop.yaml"} {
-		tt := tt
-		t.Run(tt, func(t *testing.T) {
-			bres, err := os.ReadFile(filepath.Join(tmpdir, tt))
+	seen := make(map[string]struct{})
+
+	require.NoError(t, filepath.Walk("fixtures", func(path string, info fs.FileInfo, err error) error {
+		fname := filepath.Base(path)
+		// ignore dirs and .pre.md files
+		if info.IsDir() || strings.HasSuffix(fname, ".pre.md") {
```

Comment:
Perhaps use `filepath.Ext()` for these?

```suggestion
		if info.IsDir() || filepath.Ext(fname) == ".pre.md" {
```

Reply:
Does not work as it would expect `.md`

Reply:
Ah! 🤦 didn't look how greedy it was.